crates/activity_indicator/src/activity_indicator.rs

```diff
@@ -175,7 +177,35 @@ impl ActivityIndicator {
             .flatten()
     }
 
+    fn pending_environment_errors<'a>(
+        &'a self,
+        cx: &'a AppContext,
+    ) -> impl Iterator<Item = (&'a WorktreeId, &'a EnvironmentErrorMessage)> {
+        self.project
+            .read(cx)
+            .shell_environment_errors(cx)
+            .filter(|(worktree_id, _)| {
+                !self
+                    .ignored_environment_error_worktrees
+                    .contains(worktree_id)
+            })
+    }
+
     fn content_to_render(&mut self, cx: &mut ViewContext<Self>) -> Option<Content> {
+        // Show if any direnv calls failed
+        if let Some((&worktree_id, error)) = self.pending_environment_errors(cx).next() {
+            return Some(Content {
+                icon: Some(
+                    Icon::new(IconName::Warning)
+                        .size(IconSize::Small)
+                        .into_any_element(),
+                ),
+                message: error.0.clone(),
+                on_click: Some(Arc::new(move |this, _| {
+                    this.ignored_environment_error_worktrees.insert(worktree_id);
+                })),
```

Comment:
Why keep separate state here? Can't you remove them from the project?

If you keep separate state here and then somehow the environment in the project gets reloaded and produces a different error, it'll stay ignored. If you remove it in the project, it pops up again.

I'd prefer to keep as little state as possible.

Reply:
I for some reason thought that I can't modify the project state here, my bad